tests/testthat/test-expr-expr.R

```diff
@@ -1976,104 +1976,79 @@ test_that("kurtosis", {
+      clip_min_float = c(2, 2, 3, NA, 2, Inf)
     )
   )
 
   # clip max
   expect_equal(
-    pl$DataFrame(!!!l)$select(
-      int_mini32 = pl$col("int")$clip(upper_bound = -.Machine$integer.max),
-      int_m2i32 = pl$col("int")$clip(upper_bound = -2L),
-      int_p2i32 = pl$col("int")$clip(upper_bound = 2L),
-      float_minf64 = pl$col("float")$clip(upper_bound = -Inf),
-      # float_NaN2f64 = pl$col("float")$clip(upper_bound = NaN),
-      float_p2f64 = pl$col("float")$clip(upper_bound = 2)
+    df$select(
+      clip_max_int = pl$col("int")$clip(upper_bound = 2),
+      clip_max_float = pl$col("float")$clip(upper_bound = 2)
     ),
     pl$DataFrame(
-      int_mini32 = r_clip_max(l$int, -.Machine$integer.max),
-      int_m2i32 = r_clip_max(l$int, -2L),
-      int_p2i32 = r_clip_max(l$int, 2L),
-      float_minf64 = r_clip_max(l$float, -Inf),
-      # float_NaN2f64 = r_clip_max(l$float, NaN),
-      float_p2f64 = r_clip_max(l$float, 2)
+      clip_max_int = c(1L, 2L, 2L, NA, -.Machine$integer.max, 2L),
+      clip_max_float = c(1, 2, 2, NA, -Inf, 2)
     )
   )
 
+  # clip min and max
   expect_equal(
-    pl$DataFrame(!!!l)$select(
-      a = pl$col("int")$clip(-.Machine$integer.max, -.Machine$integer.max + 1L),
-      b = pl$col("int")$clip(-2L, -2L + 1L),
-      c = pl$col("int")$clip(2L, 2L + 1L),
-      d = pl$col("float")$clip(-Inf, 1),
-      e = pl$col("float")$clip(2, 2 + 1)
+    df$select(
+      clip_int = pl$col("int")$clip(2, 3),
+      clip_float = pl$col("float")$clip(2, 3)
     ),
     pl$DataFrame(
-      a = r_clip(l$int, -.Machine$integer.max, -.Machine$integer.max + 1L),
-      b = r_clip(l$int, -2L, -2L + 1L),
-      c = r_clip(l$int, 2L, 2L + 1L),
-      d = r_clip(l$float, -Inf, 1),
-      e = r_clip(l$float, 2, 2 + 1)
+      clip_int = c(2L, 2L, 3L, NA, 2L, 3L),
+      clip_float = c(2, 2, 3, NA, 2, 3)
     )
   )
 
   # clip() accepts strings as column names
-  df <- pl$DataFrame(
-    foo = c(-50L, 5L, NA_integer_, 50L),
-    bound = c(1, 10, 1, 1)
-  )
   expect_equal(
-    df$select(clipped = pl$col("foo")$clip(lower_bound = "bound")),
-    pl$DataFrame(clipped = c(1L, 10L, NA_integer_, 50L))
+    df$select(
+      clip_min_int = pl$col("int")$clip(lower_bound = "bound"),
+      clip_min_float = pl$col("float")$clip(lower_bound = "bound")
+    ),
+    pl$DataFrame(
+      clip_min_int = c(1L, 3L, 3L, NA, 2L, .Machine$integer.max),
+      clip_min_float = c(1, 3, 3, NA, 2, Inf)
+    )
   )
 
+  # TODO-REWRITE: this should error instead of panick
+  # expect_snapshot(
+  #   df$select(pl$col("float")$clip(10, 1)),
+  #   error = TRUE
+  # )
+
+  # TODO-REWRITE: this should work instead of panick
+  # expect_equal(
+  #   df$select(
+  #     clip_min_int = pl$col("int")$clip(NaN),
+  #     clip_min_float = pl$col("float")$clip(NaN)
+  #   ),
+  #   pl$DataFrame(
+  #     clip_min_int = c(1:3, NA, -.Machine$integer.max, .Machine$integer.max),
+  #     clip_min_float = c(1, 2, 3, NA, -Inf, Inf)
+  #   )
+  # )
```

Comment:
It seems that these are only panic for dev profiles, not for release profiles. In other words, there is no difference with Python Polars.
Simply deleting the test would not be a problem.

Reply:
> It seems that these are only panic for dev profiles, not for release profiles

This sounds to me it's unexpected and should be solved, instead of deleting the tests. Did you reproduce the same panick in dev profile with py-polars?

Reply:
> This sounds to me it's unexpected and should be solved, instead of deleting the tests. Did you reproduce the same panick in dev profile with py-polars?

I haven't tried that, but building the R package with the release profile does not cause panic.
There is no way to know during testing which profile it was built with, and since the test will most likely be run in the dev profile, there is no point in testing.

Reply:
But the question is why does it fail only in the dev profile? Do you have an idea about that? I'd like to keep those tests to ensure the behavior doesn't change later on (either by choice or by mistake from upstream polars)

Reply:
If you really want to test it, I suggest you create a reproducible example using only Rust and not from Python or R to see how it works with the dev and release profiles. Simply removing the test would suffice as I don't think there is that much value in this.

Reply:
Tracked here: https://github.com/pola-rs/polars/issues/19692